pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/NoDictionarySingleColumnGroupKeyGenerator.java

```diff
@@ -122,6 +137,88 @@ public void generateKeysForBlock(TransformBlock transformBlock, int[] groupKeys)
     }
   }
 
+  public void generateKeysForBlockNullHandlingEnabled(TransformBlock transformBlock, int[] groupKeys,
+      RoaringBitmap nullBitmap) {
+    assert nullBitmap != null;
+    BlockValSet blockValSet = transformBlock.getBlockValueSet(_groupByExpression);
+    int numDocs = transformBlock.getNumDocs();
+
+    switch (_storedType) {
+      case INT:
+        int[] intValues = blockValSet.getIntValuesSV();
+        if (nullBitmap.getCardinality() < numDocs) {
+          for (int i = 0; i < numDocs; i++) {
+            groupKeys[i] = nullBitmap.contains(i) ? getKeyForNullValue() : getKeyForValue(intValues[i]);
+          }
+        } else if (numDocs > 0) {
+          Arrays.fill(groupKeys, getKeyForNullValue());
```

Comment:
Same for other places
```suggestion
          Arrays.fill(groupKeys, getKeyForNullValue(), 0, numDocs);
```

Reply:
you mean: `Arrays.fill(groupKeys, 0, numDocs, getKeyForNullValue());`